### DISCUSSION TITLE
Other downsampling resolutions (like 30 min, 2h) in Compactor

### DISCUSSION THREAD

**Discussion by santoshmungle** [thanos-io/thanos#3764] [Ideas & Features]
Hi, 

It seems that compactor only support three downsampling resolutions "raw", "5m", and "1h". For example, When I tried to set the downsampling resolution "30m" with the flag "--retention.resolution-30m", I get the error "unknown long flag" as "30m" resolution is not supported which I see in compact.go

In ideal case, any downsampling resolution should be allowed by compactor because some use cases would require different resolution in downsampling. For example, in my case, I would want to have "1d" (1 day) downsampling resolution if I want to have yearly graph of specific time-series metric. 

Is this going to be implemented in future such that any downsampling resolution would be supported by compactor? 
How one would go about having for example "30m" or "1d" downsampling resolutions supported by compactor? I guess it would require modifying compact.go and downsample.go.

**Comment by vibgy** [comment]
Although I understand your request and would like to know if its easily possible to change the downsampling resolutions.. Just note that you maybe confusing `retention.resolution` with downsampling resolution. They are two different things.
